spec/lib/search_logic_spec.rb

```diff
@@ -0,0 +1,98 @@
+RSpec.describe SearchLogic do
+    it "saves new attractions" do
+        geocode_stub = stub_request(:get, %r{https://api.geoapify.com/v1/geocode/search\?apiKey=[^&]+&format=json&text=Venice,%20Italy}).to_return(status: 200, body: File.read("spec/responses/geocode_venice.json"))
+        
+        places_stub = stub_request(:get, venice_request_url("tourism.attraction")).to_return(status: 200, body: File.read("spec/responses/attractions_venice.json"))
+
+        expect {
+            result = SearchLogic.get_places_from_city_string("Venice, Italy", "attractions")
+            expect(result).to all(be_a(Attraction))
+        }.to change(City, :count).by(1).and change(Attraction, :count).by(20)
+
+        expect(geocode_stub).to have_been_requested.times(1)
+        expect(places_stub).to have_been_requested.times(1)
+    end
+
+    it "saves new hotels" do
+        geocode_venice
```

Comment:
Don't forget to assign the stub and replace the `expect(geocode_venice)` in the other scenarios too.